importer/NoNo.py

```diff
@@ -1,53 +1,102 @@
 from Monument import Monument
 import importer_utils as utils
 
+MAPPING_DIR = "mappings"
+
 
 class NoNo(Monument):
-    """
-    A class to process monuments_no_(no).
 
-    TODO.
-    there's an API:
-    https://data.norge.no/data/riksantikvaren/kulturminnes%C3%B8k
-    look into how we can benefit from it!
-    """
+    def update_descriptions(self):
+        """
+        Add descriptions in several language.
 
-    def update_labels(self):
+        In the format "$type in $municipality, ($country if not nb)".
         """
-        Set the labels.
+        category = self.kategori.lower()
+        municip = self.get_municip_name()
+        base_norwegian = "{} i {}"
+        desc_norwegian = base_norwegian.format(category, municip)
+        self.add_description("nb", desc_norwegian)
+
+        base_english = "cultural property in {}, Norway"
+        desc_english = base_english.format(municip)
+        self.add_description("en", desc_english)
 
-        NOTE
-        Some of these are in all caps or have multiple spaces:
-        UTSIRA FYRSTASJON
-        SØGARD FJONE  -  FJONE SØNDRE
-        VÅLE PRESTEGÅRD, museum
+        base_swedish = "kulturarv i {}, Norge"
+        desc_swedish = base_swedish.format(municip)
+        self.add_description("sv", desc_swedish)
 
-        TODO
-        *Normalize - to title case?
-            It contains some old-style numbers, and these will be broken:
-            XXVIII -> Xxviii
+    def update_labels(self):
+        """
+        Add label in Bokmål.
 
-        * rm extra whitespace
+        Some of self.navn are in all caps.
+        These are normalized:
+        FÆRDER FYRSTASJON -> Færder Fyrstasjon
         """
-        for part in self.navn.split(" "):
-            if part.isupper():
-                print(part, "----", part.capitalize())
+        name = self.navn
+        if name.isupper():
+            name = name.title()
+        self.add_label("nb", name)
 
     def set_no(self):
-        self.add_statement("norwegian_monument_id", self.id)
+        """Add the Norwegian monument ID P758."""
+        self.add_statement("norwegian_monument_id", str(self.id))
+
+    def get_municip_name(self):
+        """Get the value of self.kommune without brackets."""
+        return utils.remove_markup(self.kommune)
+
+    def set_adm_location(self):
+        """Set the adm location based on kommunenummer and offline file."""
+        all_codes = self.data_files["municipalities"]
+        if self.has_non_empty_attribute("kommunenr"):
+            municip_code = str(self.kommunenr)
+            if len(municip_code) == 3:
```

Comment:
You can use `zfill(4)` for this e.g. `municip_code = str(self.kommunenr).zfill(4)`